manager/src/main/java/io/gingersnapproject/search/QueryHandler.java

```diff
@@ -24,9 +24,7 @@ public Uni<QueryResult> query(String query) {
             // Should we allow concurrency?
             .concatenate();
 
-      return result
-            .onItem()
-            .transform(searchResult -> new QueryResult(searchResult.hitCount(), searchResult.hitCountExact(),
-                  hits, searchResult.hitsExact()));
+      return Uni.combine().all().unis(result, hits.collect().asList()).combinedWith((searchResult, strings) ->
+            new QueryResult(searchResult.hitCount(), searchResult.hitCountExact(), strings, searchResult.hitsExact()));
```

Comment:
This code should probably be
```java
   public Uni<QueryResult> query(String query) {
      Uni<SearchResult> result = searchBackend.get().query(query);
      return result.onItem().transformToUni(sr ->
         Multi.createFrom().iterable(sr.hits())
               .onItem().transformToUni(sh -> caches.get(sh.indexName(), sh.documentId()))
               .concatenate()
               .collect().asList()
               .onItem().transform(list -> new QueryResult(sr.hitCount(), sr.hitCountExact(), list, sr.hitsExact()))
      );
   }
```
as it describes the relationship a bit better. The combine operator is more for separate Unis, where as the hits are not available until the result Uni has completed which is more clear with this code.

Reply:
Much better thanks. I'm applying it.